app/app_mobile_and_test.go

```diff
@@ -0,0 +1,31 @@
+// +build !ci
+
+// +build android
+
+package app
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_RootConfigDir(t *testing.T) {
+	oldEnv := os.Getenv("FILESPATH")
+	os.Setenv("FILESPATH", "/tmp")
+
+	assert.Equal(t, "/tmp", rootConfigDir())
+	os.Setenv("FILESPATH", oldEnv)
+}
+
+func Test_StoragePath(t *testing.T) {
+	oldEnv := os.Getenv("FILESPATH")
+	os.Setenv("FILESPATH", "/tmp")
+
+	p := &preferences{}
+	expected := "/tmp/storage/preferences.json"
+	assert.Equal(t, expected, p.storagePath())
+
+	os.Setenv("FILESPATH", oldEnv)
```

Comment:
use `defer` to place this near to the fetch of `oldEnv`.

Reply:
cunning :)